packages/react-test-renderer/src/ReactTestRendererScheduling.js

```diff
@@ -31,8 +31,39 @@ export function setNowImplementation(implementation: () => number): void {
   nowImplementation = implementation;
 }
 
-export function flushAll(): Array<mixed> {
-  yieldedValues = null;
+function verifyExpectedValues(expectedValues: Array<mixed>): void {
+  for (let i = 0; i < expectedValues.length; i++) {
+    const expectedValue = `"${(expectedValues[i]: any)}"`;
+    const yieldedValue =
+      i < yieldedValues.length ? `"${(yieldedValues[i]: any)}"` : 'nothing';
+    if (yieldedValue !== expectedValue) {
+      const error = new Error(
+        `Flush expected to yield ${(expectedValue: any)}, but ${(yieldedValue: any)} was yielded`,
+      );
+      // Attach expected and yielded arrays,
+      // So the caller could pretty print the diff (if desired).
+      (error: any).expectedValues = expectedValues;
+      (error: any).actualValues = yieldedValues;
+      throw error;
+    }
+  }
+
+  if (expectedValues.length !== yieldedValues.length) {
```

Comment:
Normally I would have put this check before the prior one, but that would change the error message in a lot of cases and I suspect the current message might be more helpful. Thoughts?

Reply:
Agreed, I like throwing the more specific error first.